### PR TITLE
Added navigation to Settings while in game

### LINKED ISSUE #8: Update room settings
Create a simple UI which will allow the room creator to alter the game settings.

### PR DESCRIPTION
**Pull request by sormys**
Added also firebase support for settingsUpdated message. 

Fixes #8

### CODE REVIEW

File: app/src/main/java/com/pokerio/app/PokerioFirebaseMessagingService.kt [hunk 1]
```diff
@@ -23,15 +25,29 @@ class PokerioFirebaseMessagingService : FirebaseMessagingService() {
...
+                    data["nickname"]!!,
+                    data["playerHash"]!!
                 )
             )
         }
+        fun settingsUpdated(data: MutableMap<String, String>) {
+            PokerioLogger.debug("Received updatedSettings FCM message")
+
+            GameState.changeGameSettings(
+                data["startingFunds"]?.toInt() ?: GameState.startingFunds,
```
Comment (Kwasow) [inline app/src/main/java/com/pokerio/app/PokerioFirebaseMessagingService.kt]: Please use non-null assertions instead, like this: `data["startingFunds"]!!`

File: app/src/main/java/com/pokerio/app/screens/SettingsScreen.kt [hunk 2]
```diff
@@ -57,6 +61,31 @@ fun SettingsScreen(
         Context.MODE_PRIVATE
     )
 
+    DisposableEffect(LocalLifecycleOwner.current) {
+        onDispose {
+            // Unregister callback when we leave the view
+            GameState.exitSettings(
+                context = context,
+                onError =
```
Comment (Kwasow) [inline app/src/main/java/com/pokerio/app/screens/SettingsScreen.kt]: I'd prefer if you did it like this:

```kotlin
val onError = { doSomething() }
val onSuccess = { doSomething() }
GameState.exitSettings(context, onSuccess, onError)
```

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 3]
```diff
@@ -15,25 +15,28 @@ import kotlinx.serialization.json.jsonArray
...
     var smallBlind: Int = -1
+    var isPlayerAdmin: Boolean = false
 
     // Callbacks
-    private var playerJoinedCallbacks = HashMap<Int, (Player) -> Unit>()
-    private var playerRemovedCallbacks = HashMap<Int, (Player) -> Unit>()
+    var onGameReset = {}
+    private val playerJoinedCallbacks = HashMap<Int, (Player) -> Unit>()
+    private val playerRemovedCallbacks = HashMap<Int, (Player) -> Unit>()
+    private var SettingsChangedCallback = HashMap<Int, () -> Unit>()
```
Comment (Kwasow) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt]: Please use lower camel case for this `settingsChangedCallback`

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 4]
```diff
@@ -58,7 +61,7 @@ object GameState {
         // TODO: Load settings
```
Comment (Kwasow) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt]: Could you take care of this TODO?

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 5]
```diff
@@ -139,6 +145,89 @@ object GameState {
...
+                    url.readText()
+                    ContextCompat.getMainExecutor(context).execute(onSuccess)
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                    PokerioLogger.error(e.toString())
+                    ContextCompat.getMainExecutor(context).execute(onError)
+                }
+            }
+        }
+    }
```
Comment (Kwasow) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt:198]: Please add a newline after this function

**Review by Kwasow — COMMENTED** [review]
I like how it works but a few style changes are worth making

File: app/src/main/java/com/pokerio/app/utils/GameState.kt (hunk 4 above)
Comment (sormys) [reply app/src/main/java/com/pokerio/app/utils/GameState.kt]: I added it now, please check if it is ok.

File: app/src/main/java/com/pokerio/app/utils/GameState.kt (hunk 3 above)
Comment (sormys) [reply app/src/main/java/com/pokerio/app/utils/GameState.kt]: Done.

File: app/src/main/java/com/pokerio/app/screens/SettingsScreen.kt (hunk 2 above)
Comment (sormys) [reply app/src/main/java/com/pokerio/app/screens/SettingsScreen.kt]: Done.

File: app/src/main/java/com/pokerio/app/PokerioFirebaseMessagingService.kt (hunk 1 above)
Comment (sormys) [reply app/src/main/java/com/pokerio/app/PokerioFirebaseMessagingService.kt]: Done.

File: app/src/main/java/com/pokerio/app/utils/GameState.kt (hunk 5 above)
Comment (sormys) [reply app/src/main/java/com/pokerio/app/utils/GameState.kt:198]: Done.

**Review by Kwasow — APPROVED** [review]

**Review by mikolaj-pirog — APPROVED** [review]

**Pull request merged**